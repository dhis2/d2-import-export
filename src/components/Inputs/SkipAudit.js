import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_YES = { value: 'true', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const SKIP_AUDIT_KEY = 'skipAudit'
export const SKIP_AUDIT_DEFAULT_VALUE = OPTION_NO.value

export const SkipAudit = () => (
    <RadioGroup
        name={SKIP_AUDIT_KEY}
        label={i18n.t('Skip audit')}
        options={[OPTION_YES, OPTION_NO]}
        defaultValue={OPTION_NO.value}
    />
)
