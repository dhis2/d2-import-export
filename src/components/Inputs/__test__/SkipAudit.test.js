import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { SkipAudit } from '../SkipAudit'

jest.mock('react-final-form')

describe('Input component - SkipAudit', () => {
    it('should render correctly', () => {
        const file = shallow(<SkipAudit />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
