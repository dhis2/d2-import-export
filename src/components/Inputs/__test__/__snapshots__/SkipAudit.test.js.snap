// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - SkipAudit should render correctly 1`] = `
<Field>
  <Label>
    Skip audit
  </Label>
  <RadioGroup
    dataTest="input-skip-audit"
    defaultValue="false"
    label="Skip audit"
    name="skipAudit"
    options={
      Array [
        Object {
          "label": "Yes",
          "value": "true",
        },
        Object {
          "label": "No",
          "value": "false",
        },
      ]
    }
  />
</Field>
`;
